src/init/entry.js

```diff
@@ -73,50 +59,40 @@ function handleCacheExpired(domElement: HTMLElement) {
     );
 }
 
-function logError(error) {
-    if (error instanceof Error) {
-        log.error(error.toString());
-    } else if (error) {
-        log.error(JSON.stringify(error));
-    }
+function runApp(store: ReduxStore, history: HashHistory) {
+    store.dispatch(loadApp());
+    addBeforeUnloadEventListener(store);
+    render(
+        <JSSProviderShell>
+            <DataProvider
+                baseUrl={process.env.REACT_APP_DHIS2_BASE_URL}
+                apiVersion=""
+            >
+                <App
+                    store={store}
+                    history={history}
+                />
+            </DataProvider>
+        </JSSProviderShell>,
+        domElement,
+    );
 }
 
-async function loadAppAsync(domElement: HTMLElement) {
+async function loadAppAsync() {
     render(
         <JSSProviderShell>
-            <LoadingMask />
+            <DataProvider
+                baseUrl={process.env.REACT_APP_DHIS2_BASE_URL}
+                apiVersion=""
+            >
+                <AppLoader
+                    onRunApp={runApp}
+                    onCacheExpired={handleCacheExpired}
+                />
+            </DataProvider>
         </JSSProviderShell>,
         domElement,
     );
-
-    try {
-        await initializeAsync(() => handleCacheExpired(domElement));
-        const history = createHistory();
-        const store = getStore(history, () => runApp(domElement, store, history));
-    } catch (error) {
-        let message = 'The application could not be loaded.';
-        if (error && error instanceof DisplayException) {
-            logError(error.innerError);
-            message += ` ${error.toString()}`;
-        } else {
-            logError(error);
-            if (process.env.NODE_ENV !== environments.prod) {
-                message += ' Please verify that the server is running.';
-            } else {
-                message += ' Please see log for details.';
-            }
-        }
-
-        render(
-            <div>{message}</div>,
-            domElement,
-        );
-    }
 }
 
-const domElement = document.getElementById(DOM_ID);
-if (!domElement) {
-    log.error(`html element with id ${DOM_ID} is missing in index file`);
-} else {
-    loadAppAsync(domElement);
-}
+loadAppAsync();
```

Comment:
Why we dont export this function so that we can call it on this [file](https://github.com/dhis2/capture-app/blob/refactor-metadata-fetching/src/index.js)?

Every person who would start reading this code from scratch and knows react would fo into that file. When you first  go there as completely new to the project, as it is now, you see just some imports and you have to look into all those files to find where the app is actually starting from. I would argue that will help people who are gonna join us in the future to better understand what is happening here. 

What is your stand here Joakim?

Reply:
Making an exception to my stay on point rule for this suggestion. Thanks 👍